### PR TITLE
Speedup Inference (to get Embeddings for training & new repo on-boarding) by 10x

### PR DESCRIPTION
**Pull request by hamelsmu**
### Created utility that speeds up generating of embeddings for GitHub issues by 10x when doing it in batch.

This will allow us to onboard new repos in a much more realistic timeframe and will be a much better user experience, especially for massive orgs like Kubernetes.

This was done by taking the following steps:

- Hacking the machinery of fastai datablock api to perform multi-threaded transformation for inference (there isn't an easy way out of the box to do this)
- Sorting Issues by their sequence lengths before generating batches to minimize the amount of padding necessary in each batch.
- Batching the issues together (previously we were performing inference on one issue at a time). 

image

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/code-intelligence/30)

### CODE REVIEW

**Comment by hamelsmu** [comment]
@googlebot I fixed it.

**Comment by abcdefgs0324** [comment]
I think googlebot checks whether all commit authors sign the CLA. So, it seems like you can not have commits signed as root..

BTW, I will check in the code you updated. Thanks!

File: Issue-Embeddings/flask_app/inference.py [hunk 1]
```diff
@@ -24,15 +31,18 @@ def __init__(self,
         self.learn.model.eval()  # turn off dropout, etc. only need to do this after loading model.
         self.encoder = self.learn.model[0]
         self.pad_idx = self.learn.data.pad_idx
+        self.mtokenizer = [x.tokenizer for x in self.learn.data.processor if type(x) == TokenizeProcessor][0]
```
Comment (jlewi) [inline Issue-Embeddings/flask_app/inference.py]: nit:  What does "m" in tokenizer stand for. We generally eschew abbreviations
http://google.github.io/styleguide/pyguide.html#316-naming

File: Issue-Embeddings/flask_app/inference.py [hunk 2]
```diff
@@ -148,31 +159,50 @@ def df_to_emb(self, dataframe:pd.DataFrame, bs=150) -> np.ndarray:
         (200, 2400)
         """
         new_df = self.process_df(dataframe)
-        text_list = new_df['text'].to_list()
-        
-        numercalized_docs = []
+        #text_list = new_df['text'].to_list()
```
Comment (jlewi) [inline Issue-Embeddings/flask_app/inference.py]: nit: can we delete this commented out code?

**Comment by jlewi** [comment]
I tried to manually override the CLA check; didn't seem to work. The easiest way to override the CLA check is to rebase and squash your commits. This will rewrite the commits with you as the author so that the CLA check passes.

e.g.

```
git fetch origin
git rebase -i origin/master
```

This should pop a window showing you a list of commits on your branch that are not on master
Select "pick" for the first commit and then "squash" for all subsequent commits. This will create a single commit with all your changes.

File: Issue-Embeddings/flask_app/inference.py (hunk 2 above)
Comment (hamelsmu) [reply Issue-Embeddings/flask_app/inference.py]: 👍

File: Issue-Embeddings/flask_app/inference.py (hunk 1 above)
Comment (hamelsmu) [reply Issue-Embeddings/flask_app/inference.py]: 👍

**Comment by hamelsmu** [comment]
@jlewi the CLA check still fails after rebase.  Shall we just ignore it this time , or should I open a separate PR?

**Comment by hamelsmu** [comment]
FYI -- I tried opening a separate PR, that didn't work

**Comment by jlewi** [comment]
I was able to manually override the CLA.

**Comment by jlewi** [comment]
@abcdefgs0324 Can you review this PR and LGTM and approve it if it looks good?

**Comment by abcdefgs0324** [comment]
@hamelsmu @jlewi  Seems like I can not approve to merge it.
